src/solver.cpp

```diff
@@ -525,6 +528,16 @@ void Solver::setSensitivityMethod(const SensitivityMethod sensi_meth) {
     this->sensi_meth = sensi_meth;
 }
 
+void Solver::setSensitivityMethodPreequilibration(const SensitivityMethod sensi_meth_preeq) {
+    if (rdata_mode == RDataReporting::residuals &&
+        sensi_meth_preeq == SensitivityMethod::adjoint)
+        throw AmiException("Adjoint Sensitivity Analysis is not compatible with"
+                           " only reporting residuals!");
+    if (sensi_meth_preeq != this->sensi_meth_preeq)
```

Comment:
Is this check really necessary? I don't think `sensi_meth_preeq` affects any memory allocation in this `Solver` instance, but only influences what happens when the SteadyStateSolver is generated, right?